### PR TITLE
enrollments for single user

### PR DESCRIPTION
**Pull request by abeglova**
#### Pre-Flight checklist
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [x] Settings
  - [x] New settings are documented and present in `app.json`
  - [x] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
closes https://github.com/mitodl/open-discussions/issues/3298

#### What's this PR do?
This pr queries for enrollments from the Athena enrollment table for a single user and creates Enrollment records in the open database. It attempts to match the course code from the Athena database (which comes from the mitx bigquery database) to a course run or, failing that, a course.

#### How should this be manually tested?
Ask me for Athena credentials. Set 
ATHENA_ACCESS_KEY_ID
ATHENA_SECRET_ACCESS_KEY
ATHENA_WORK_GROUP
and
ATHENA_REGION_NAME
in your .env file

1) Verify that you can still create a new user through the UI and that you see `Task course_catalog.tasks.update_enrollments_for_email` run on celery after you set the password.

2) Change the email of the user you just created to `pdpinch@mit.edu` Run

```
from course_catalog.tasks import *
update_enrollments_for_email.delay('pdpinch@mit.edu')
```
Verify that 154 enrollments are created

### CODE REVIEW

File: course_catalog/admin.py
Comment (pdpinch) [inline course_catalog/admin.py:170]: What changes are you making the admin here?

File: requirements.txt
Comment (pdpinch) [inline requirements.txt:147]: small thing: is there anyway to get pip-compile to maintain the line breaks, so this diff isn't so huge?

File: course_catalog/admin.py
Comment (abeglova) [reply course_catalog/admin.py:170]: I'm making changes to the Enrollments admin page (currently no data).

There aren't matches for runs for all enrollments. 
This updates the admin page to display just the course match for courses where there is a match on the course id but not the run id. It also shows the run id from the enrollments table

File: app.json
Comment (blarghmatey) [inline app.json:42]: We shouldn't be adding separate access keys for Athena, we should just add the appropriate permissions to the AWS credentials that the application is already using.

File: course_catalog/etl/enrollments.py
Comment (blarghmatey) [inline course_catalog/etl/enrollments.py:47]: I think it would make sense to pass the cursor as a parameter to this function so that it is easier to swap to Trino from Athena once we have that deployed. It would also simplify mocking the cursor object for tests.

File: course_catalog/etl/enrollments.py
Comment (blarghmatey) [inline course_catalog/etl/enrollments.py:34]: It seems like this function shouldn't have any awareness of the settings requirements for the cursor object and should be external to the logic to process the query results.

File: app.json
Comment (abeglova) [reply app.json:42]: Agreed! Can you give the rc credentials for open access to Athena?

File: course_catalog/etl/enrollments.py
Comment (abeglova) [reply course_catalog/etl/enrollments.py:34]: I was using the credentials as a feature flag so we don't get noisy errors if this is merged before the credentials are set. If we are going to just use the existing AWS credentials, this isn't necessary. The only one of these credentials that we will keep is ATHENA_WORK_GROUP, and 'primary' will be the default so it will work without any credentials set

File: course_catalog/etl/enrollments.py
Comment (abeglova) [reply course_catalog/etl/enrollments.py:47]: Once we remove the check for credentials above, swapping the cursor will just involve changing athena_cursor()
 to trino_curosr() here. Passing the cursor to this functions works too though

File: course_catalog/etl/enrollments.py
Comment (mbertrand) [inline course_catalog/etl/enrollments.py:73]: Is it okay to create an enrollment if both `run` and `course` are `None`?  Should there be an error logged if `course` is also `None` to notify someone that the Athena `enrollment_course_id` doesn't have any match in open?

File: course_catalog/tasks_test.py
Comment (mbertrand) [inline course_catalog/tasks_test.py:493]: Maybe also add a test to verify that the task is not called if no user is found with a matching email address

```
    update_enrollments_for_email.delay(fake_email)
    mock_task.assert_not_called()
```

**Review by mbertrand — CHANGES_REQUESTED** [review]

File: requirements.txt
Comment (abeglova) [reply requirements.txt]: fixed

File: course_catalog/etl/enrollments.py
Comment (abeglova) [reply course_catalog/etl/enrollments.py:73]: There are enough courses without matches that raising an error does not make sense. Once this is merged and the data is updated, the plan is to review the enrollment_course_ids with no matches to see if there are any we care about then add a look up table to match them to the correct course id

File: app.json
Comment (blarghmatey) [reply app.json:42]: Please open a ticket for that, preferably at https://trello.com/b/jLF7h6ZW/platform-engineering-open-learning-engineering

**Review by mbertrand — COMMENTED** [review]
Code looks great, but I'm now getting an AWS AccessDenied error probably because the CI AWS key I'm using doesn't have the correct permissions yet.  Probably should hold off on merging until those permissions are in place?


```

File: app.json
Comment (abeglova) [reply app.json]: I don't have access to that trello board

Comment (abeglova) [reply app.json]: Here is an open issue https://github.com/mitodl/open-discussions/issues/3318

File: app.json
Comment (blarghmatey) [inline app.json:49]: As part of setting up the permissions for Open to access Athena I have set up per-environment workgroups and database names, so there should be an additional parameter of `"ATHENA_DATABASE_NAME"` that is used to determine which database to execute queries against.

File: course_catalog/etl/enrollments.py
Comment (blarghmatey) [inline course_catalog/etl/enrollments.py:33]: This line needs to be configurable to the database name that will be queried.

File: course_catalog/etl/enrollments.py
Comment (blarghmatey) [inline course_catalog/etl/enrollments.py]: This should be `settings.ATHENA_MITX_DATABASE_NAME`. We might also want to make the table name configurable as it will likely evolve/change as we add more enrollments data sources.

**Review by blarghmatey — APPROVED** [review]

**Comment by blarghmatey** [comment]
:+1:

**Pull request merged**
